opentracing/src/main/java/opentracing/Tracer.java

```diff
@@ -13,25 +13,41 @@
  */
 package opentracing;
 
+
 /**
- * Tracer is a simple, thin interface for Span creation.
+ * Tracer is a simple, thin interface for Span creation, and Span propagation into different transport formats.
  */
 public interface Tracer {
 
   /**
-   * Create, start, and return a new Span with the given `operationName`, all without specifying a
-   * parent Span that can be used to incorporate the newly-returned Span into an existing trace.
+   * Create, start, and return a new Span with the given `operationName`.
+   * An optional parent Span can be specified used to incorporate the newly-returned Span into an existing trace.
    *
    * <p>Example:
    * <pre>{@code
    * Tracer tracer = ...
-   * Span feed = tracer.startTrace("GetFeed");
-   * Span http = tracer.startTrace("HandleHTTPRequest")
+   * Span feed = tracer.startTrace("GetFeed", null);
+   * Span http = tracer.startTrace("HandleHTTPRequest", feed)
    *                   .setTag("user_agent", req.UserAgent)
    *                   .setTag("lucky_number", 42);
    * }</pre>
    */
-  Span startTrace(String operationName);
+  Span startSpan(String operationName, /* @Nullable */ Span parent);
+
+  /** Takes two arguments:
+   *    a Span instance, and
+   *    a “carrier” object in which to inject that Span for cross-process propagation.
+   *
+   */
+  <T> void inject(Span span, T carrier);
```

Comment:
If this is a war of attrition then I give up. :)

If you don't want to support the format parameter, let's just get this in without it and we can write concrete, non-templated carrier types rather than passing ambiguous `byte[]` buffers to inject/join (which will not work in practice).

PS: you wrote...

> Note that it was not me that advocated for the removal of those intermediate methods, for a smaller api simplification that what the delegation model here provides.

The intermediate methods (and types) are not relevant here... the question is whether runtime type information alone is sufficient to identify a carrier. The intermediate layer of Injector/Extractor interfaces did not make that any easier or harder so I'm not sure why it's being brought up here.